ding/policy/base_policy.py

```diff
@@ -421,10 +421,22 @@ def sync_gradients(self, model: torch.nn.Module) -> None:
             gradients allreduce and optimizer updates.
         """
 
+        # if self._bp_update_sync:
+        #     for name, param in model.named_parameters():
+        #         if param.requires_grad:
+        #             if param.grad is not None:
+        #                 allreduce(param.grad.data)
+        # else:
+        #     synchronize()
         if self._bp_update_sync:
             for name, param in model.named_parameters():
                 if param.requires_grad:
-                    allreduce(param.grad.data)
+                    if param.grad is not None:
+                        allreduce(param.grad.data)
+                    else:
+                        # 如果梯度为 None，则创建一个与 param.grad_size 相同的零张量，并执行 allreduce
```

Comment:
remove commented code and add English comment, then these modifications will be merged

Reply:
done